examples/debugging/simple_spread/feedforward/decentralised/run_maddpg_scale_trainers.py

```diff
@@ -0,0 +1,131 @@
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Example running feedforward MADDPG on debug MPE environments."""
+
+import functools
+from datetime import datetime
+from typing import Any
+
+import launchpad as lp
+import sonnet as snt
+from absl import app, flags
+from launchpad.nodes.python.local_multi_processing import PythonProcess
+
+from mava.systems.tf import maddpg
+from mava.systems.tf.maddpg import make_default_networks
+from mava.utils import lp_utils
+from mava.utils.environments import debugging_utils
+
+FLAGS = flags.FLAGS
+flags.DEFINE_string(
+    "env_name",
+    "simple_spread",
+    "Debugging environment name (str).",
+)
+flags.DEFINE_string(
+    "action_space",
+    "continuous",
+    "Environment action space type (str).",
+)
+flags.DEFINE_string(
+    "mava_id",
+    str(datetime.now()),
+    "Experiment identifier that can be used to continue experiments.",
+)
+flags.DEFINE_string("base_dir", "~/mava/", "Base dir to store experiments.")
+
+
+def main(_: Any) -> None:
+
+    # environment
+    environment_factory = functools.partial(
+        debugging_utils.make_environment,
+        env_name=FLAGS.env_name,
+        action_space=FLAGS.action_space,
+    )
+
+    # networks
+    network_factory = lp_utils.partial_kwargs(make_default_networks)
+
+    # loggers
+    # set custom logger config for each process
+    # -- log trainer,executor and evaluator to TF
+    # -- log only evaluator to terminal
+    log_every = 10
+    logger_config = {
+        "trainer": {
+            "directory": FLAGS.base_dir,
+            "to_terminal": False,
+            "to_tensorboard": True,
+            "time_stamp": FLAGS.mava_id,
+            "time_delta": log_every,
+        },
+        "executor": {
+            "directory": FLAGS.base_dir,
+            "to_terminal": False,
+            "to_tensorboard": True,
+            "time_stamp": FLAGS.mava_id,
+            "time_delta": log_every,
+        },
+        "evaluator": {
+            "directory": FLAGS.base_dir,
+            "to_terminal": True,
+            "to_tensorboard": True,
+            "time_stamp": FLAGS.mava_id,
+            "time_delta": log_every,
+        },
+    }
+
+    # Checkpointer appends "Checkpoints" to checkpoint_dir
+    checkpoint_dir = f"{FLAGS.base_dir}/{FLAGS.mava_id}"
+
+    # distributed program
+    program = maddpg.MADDPG(
+        environment_factory=environment_factory,
+        network_factory=network_factory,
+        logger_config=logger_config,
+        num_executors=2,
+        shared_weights=False,
+        trainer_networks={
+            "trainer_0": ["agent_0"],
+            "trainer_1": ["agent_1"],
+            "trainer_2": ["agent_2"],
+        },
+        executor_samples=[["agent_0", "agent_1", "agent_2"]],
```

Comment:
Why is this called `executor_samples`?

Reply:
Executor samples are the network samples that are chosen at random to construct an agent pool at the start of an episode. In the vanilla case, there will be only one network configuration to sample from. But this also allows for more complex sampling. In the case above the executor will sample networks randomly one at a time until all the agent slots are filled. Then at the end of the episode, the experience gets sent to the trainers corresponding to the networks that were used. I can explain it in more detail in a presentation.

Reply:
@DriesSmit As discussed, I think we need to simplify the names of these variables so that it is easier to tell what they are doing.

Reply:
What names do you think I should use? Maybe `network_sample_sets`?